zerosum/zerosum.py

```diff
@@ -0,0 +1,38 @@
+"""
+ID: whuan2001
+LANG: PYTHON2
+TASK: zerosum
+"""
+def solution():
+    fin = open("zerosum.in", "r")
+    fout = open("zerosum.out", "w")
+    N = int(fin.readline().strip())
+    nums = []
+    res = []
+    nums_links = ""
+    calculated_nums = []
+    dfs(N, 1, 1, calculated_nums, nums_links, res)
+    res = sorted(res)
+    for r in res:
+        print >>fout, r
+def dfs(N, cur_num, num, calculated_nums, nums_links, res):
+    if num == N:
+        if sum(calculated_nums) + cur_num == 0:
```

Comment:
You don't need a list of numbers to calculate, you can add them one by one during recursion. You don't need a list.